constantine/eth_verkle_ipa/transcript_gen.nim

```diff
@@ -27,37 +29,72 @@ func newTranscriptGen*(res: var CryptoHash, label: openArray[byte]) =
   res.update(label)
 
 func messageAppend*(res: var CryptoHash, message: openArray[byte], label: openArray[byte]) =
-  res.init()
   res.update(label)
   res.update(message)
 
 func messageAppend_u64*(res: var CryptoHash, label: openArray[byte], num_value: uint64) = 
-  res.init()
   res.update(label)
   res.update(num_value.toBytes(bigEndian))
 
 func domainSeparator*(res: var CryptoHash, label: openArray[byte]) =
-  var state {.noInit.} : CryptoHash
-  state.update(label)
+  res.update(label)
 
 func pointAppend*(res: var CryptoHash, label: openArray[byte], point: EC_P) =
   var bytes {.noInit.}: array[32, byte]
-  if bytes.serialize(point) == cttCodecEcc_Success:
-    res.messageAppend(bytes, label)
+  let stat = bytes.serialize(point) 
+  doAssert stat == cttCodecEcc_Success, "Serialization Failure!"
```

Comment:
```suggestion
  let status {.used.} = bytes.serialize(point)
  debug: doAssert status == cttCodecEcc_Success, "transcript_gen.pointAppend: Serialization Failure!"
```
At the moment serialiation of elliptic curve points cannot fail.

https://github.com/mratsim/constantine/blob/58d8d2c62256dfc4e347e049de102abacee5f548/constantine/serialization/codecs_banderwagon.nim#L47-L74

The `bool` has been added in anticipation of having a procedures relying of generic serialize (or serializing concept), but here it can just be a debug-only assert